File: 1346-Check-If-N-and-Its-Double-Exist.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var checkIfExist = function(arr) {
  for (const i in arr) {
    const x = arr.lastIndexOf(arr[i] / 2)
    if (x != -1 && x != i) return true;
  }
    
  return false;
};

File: solutions/1346-Check-If-N-and-Its-Double-Exist.cpp
class Solution {
public:
  bool checkIfExist(vector<int>& arr) {
    unordered_map<int, int> map;
    for (auto it = arr.begin(); it != arr.end(); it++) {
      if (map.find((*it) * 2) != map.end()) return true;
      if(map.find((*it) / 2) != map.end() and (*it) % 2 == 0) return true; 
      map[*it] = 1;
    }
    
    return false;
  }
};
